### DISCUSSION TITLE
blog plugin: .authors.yml - avatar - relative path for local resource

### DISCUSSION THREAD

**Discussion by AmebaBrain** [squidfunk/mkdocs-material#7211] [Q&A] [answered]
It looks like for the `blog` plugin, in `.authors.yml` file the `avatar` attribute must be specified as an **url** to the existing resource. Because when user is displayed in different places - list of posts, filter by category, etc. it's referenced by different dynamically generated urls.

In sources for the blog of the `mkdocs-material` itself they are also set as an absolute url

https://github.com/squidfunk/mkdocs-material/blob/09f817d1541bd20cd721dacf224547a44aebce2e/docs/blog/.authors.yml#L1-L6

If you try to refer to a local resource, as when you refer to other sibling files from within markdown, you get an error, when the user info is displayed in a different context/view. 
```
WARNING -  [16:26:52] "GET /blog/archive/2022/assets/images/author_thumbnail.jpeg HTTP/1.1" code 404  
```
This happens because file is actually inside `assets/images` folder. But because use currently opened one of the Archive index pages, `mkdocs-material` dynamically preprends the path to the resource with `blog/archive/2022/`.

**Comment by kamilkrzyskow** [comment]
Hello @AmebaBrain,
I'm not sure what is your question? This is a duplicate of:
- https://github.com/squidfunk/mkdocs-material/discussions/5769

Yes, the resource needs to be an URL, not a relative path.

**Reply by AmebaBrain** [reply]
Yes, thanks for the exact pointer. Question is that to have it displayed correctly - you must refer to some resource available over the `http(s)`. So, you have external dependency, even during local development for `mkdocs serve`.

Website built with `mkdocs` acts as a SPA. And any SPA allows to refer to local resource. I'm working on a new website which is not yet deployed. And during development I must specify an url to the image available over the internet..it's not handy.

I see that there was a general agreement to make it resolvable from relative path.
![image](https://github.com/squidfunk/mkdocs-material/assets/16878997/efccdb5f-77ea-4f5b-a956-8d4ff5149031)

**Reply by squidfunk** [reply, marked as answer]
In the screenshot you posted I asked for a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) in a new, proper, issue. Until now, nobody took the time to create one, so it doesn't seem that this is a big problem for users. If it is, investing a little time to make it easier for us to fix or implement it is a fair ask, from a maintainers perspective ✌️

**Comment by AmebaBrain** [comment]
@squidfunk , I've created a Feature request #7230 and attached to it reproduction archive. Attaching here as well, just in case.
[9.5.25-post-author-display-bug.zip](https://github.com/user-attachments/files/15519682/9.5.25-post-author-display-bug.zip)

Many thanks for a great project!
